src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -1589,12 +1604,23 @@ def aks_enable_addons(cmd, client, resource_group_name, name, addons, workspace_
                       subnet_name=None, no_wait=False):
     instance = client.get(resource_group_name, name)
     subscription_id = _get_subscription_id(cmd.cli_ctx)
-
+    service_principal_client_id = instance.service_principal_profile.client_id
     instance = _update_addons(cmd, instance, subscription_id, resource_group_name, addons, enable=True,
                               workspace_resource_id=workspace_resource_id, subnet_name=subnet_name, no_wait=no_wait)
 
     if 'omsagent' in instance.addon_profiles:
         _ensure_container_insights_for_monitoring(cmd, instance.addon_profiles['omsagent'])
```

Comment:
based on my understanding of the code, this will be invoked only when monitoring add-on is enabled. Is that right? If so, this will be a weird experience for users who have already onboarded to monitoring. Is that scenario not supported via the CLI? 
We have three scenarios
1. Brownfield with monitoring enabled (give permissions) 
2. Brownfield with no monitoring enabled( onboardi + give permissions) 
3. Greenfield

Reply:
CLI and all other clients, we will be adding the role assignment only when the user selects the monitoring addon.  For the existing customers, we will be enabling the experience through the banner and scripts, and eventually, this will be enabled from Ux without manual steps.